src/indexer/allo/v2/handleEvent.ts

```diff
@@ -452,21 +514,29 @@ export async function handleEvent(
         throw new Error("Round not found");
       }
 
-      // TODO: discuss how to handle differnt decode based on round.strategyName
-      const values = decodeDVMDApplicationData(encodedData);
+      let id;
+      let values;
 
-      const metadata = await ipfsGet<ApplicationTable["metadata"]>(
-        values.metadata.pointer
-      );
+      switch (round.strategyName) {
+        case "allov2.DirectGrantsSimpleStrategy":
+          values = decodeDGApplicationData(encodedData);
+
+          id = event.params.recipientId;
+          break;
+
+        case "allov2.DonationVotingMerkleDistributionDirectTransferStrategy":
+        default:
```

Comment:
we should throw an error if the strategy is not recognized instead of doing something that might not make sense and cause other problems